ethereum/stratum/src/main/java/org/hyperledger/besu/ethereum/stratum/StratumServer.java

```diff
@@ -126,6 +158,12 @@ public void newJob(final PoWSolverInputs poWSolverInputs) {
     for (StratumProtocol protocol : protocols) {
       protocol.setCurrentWorkTask(poWSolverInputs);
     }
+    UInt256 difficulty = poWSolverInputs.getTarget().divide(divisor);
+    if (!difficulty.fitsLong()) {
```

Comment:
Prometheus uses floating points in the transport.  Why trim to long when we could just convert to Double?

Reply:
We will be in floating point quite a bit before Long.MAX_VALUE, it only seems to get 9 digits of precision.

Reply:
We don't just support Prometheus anymore, we have OpenTelemetry now ❤️ 
I can switch to Double to make sure the value always fits, good point.